### ISSUE TITLE
[Transform] retry searches despite skip_unavailable is set to true for a CCS remote

### ISSUE THREAD

**Issue by hendrikmuhs** [elastic/elasticsearch#84090]
Transform by design retries if search returns partial results to not create results for incomplete data. CCS offers a similar setting called `skip_unavailable`. However this is not part of the search DSL but a cluster setting. A discussion about this choice can be found in #82407.

We've discovered this issue recently for data feeds: #83838. Apart from transform rollup(v1) is probably affected, too.

**Mitigation**

Do not use a remote connection that uses `skip_unavailable`. In case you need to set `skip_unavailable` to `true`, use separate remote connections.

**Fix**

If the current design stays, we need a similar fix as done in #84052. If the fix is applied in the super classes it is not necessary to have a separate fix for rollup(v1).

**Comment by hendrikmuhs** [comment]
Note: Transform uses a `pit` handle, this might have additional side effects and should be investigated as part of the fix.

**Comment by benwtrent** [comment]
Blocked by: https://github.com/elastic/elasticsearch/issues/84400
